fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -506,11 +522,9 @@ private void handleShowDb() throws AnalysisException {
                 continue;
             }
 
-            if (!GlobalStateMgr.getCurrentState().getAuth().checkDbPriv(ConnectContext.get(), fullName,
-                    PrivPredicate.SHOW)) {
+            if (!GlobalStateMgr.getCurrentState().getAuth().checkDbPriv(ConnectContext.get(), fullName, PrivPredicate.SHOW)) {
```

Comment:
@adzfolc Is it work for external databases?

Reply:
yes, it works

Reply:
> @adzfolc Is it work for external databases?

MySQL [demodb]> show databases from catalog_1;
+----------+
| Database |
+----------+
| default  |
| tn_test  |
| un_test  |
+----------+
3 rows in set (0.27 sec)

MySQL [demodb]> show databases from catalog_1 like 'default';
+----------+
| Database |
+----------+
| default  |
+----------+
1 row in set (0.00 sec)

MySQL [demodb]> show databases from catalog_1 like 'tn_test';
+----------+
| Database |
+----------+
| tn_test  |
+----------+
1 row in set (0.08 sec)

MySQL [demodb]> show databases from catalog_1 like '%test';
+----------+
| Database |
+----------+
| tn_test  |
| un_test  |
+----------+
2 rows in set (0.00 sec)

MySQL [demodb]> show databases from catalog_1 like 'aa';
Empty set (0.01 sec)